SwanCustomEnvironments/swancustomenvironments/serverextension.py

```diff
@@ -0,0 +1,88 @@
+from tornado import web
+from traitlets import Unicode
+from traitlets.config import Configurable
+
+from jupyter_server.base.handlers import JupyterHandler, APIHandler
+from jupyter_server.utils import url_path_join
+
+import subprocess
+import re
+from os import path
+
+
+class SwanCustomEnvironments(Configurable):
+
+    makenv_path = Unicode(
+        path.join(path.dirname(__file__), "scripts/makenv.sh"),
+        config=True,
+        help='Path to the script used to create custom environments.'
+    )
+
+
+class SwanCustomEnvironmentsApiHandler(APIHandler):
+
+    config = None
+
+    def initialize(self):
+        self.config = SwanCustomEnvironments(config=self.config)
+
+    @web.authenticated
+    def get(self):
+        self.set_header("Content-Type", "text/event-stream")
+
+        env_name = self.get_query_argument("env", default=None)
+        requirements = self.get_query_argument("req", default=None)
+        clear = self.get_query_argument("clear", default="true")
+        accpy_version = self.get_query_argument("accpy", default=None)
+
+        try:
+            arguments = ["--env", env_name, "--req", requirements]
+            if clear.lower() == "true":
+                arguments.extend(["--clear"])
+            if accpy_version is not None:
+                arguments.extend(["--accpy", accpy_version])
+            
+            makenv_process = subprocess.Popen([self.config.makenv_path, *arguments], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+            
+            for line in iter(makenv_process.stdout.readline, b""):
+                self.write(f"data: {line.decode('utf-8')}\n\n")
+                self.flush()
+            
+        except Exception as e:
```

Comment:
Can you be more specific about the type of exceptions the code above can raise? For example, if `subprocess.Popen` can raise a different exception than `self.write`, you can have two `except` blocks, each with its exception type and its more specific error message.